### PR TITLE
Update LwM2M Client STM guide with regards to SBSFU

### PR DESCRIPTION
**Pull request by jgoreckiavs**

### CODE REVIEW

**Comment by jgoreckiavs** [comment]
@JZimnol  @malw4 @Kucmasz please review

File: docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md [hunk 1]
```diff
@@ -4,15 +4,15 @@ Integrate your P-L496G-CELL02 Discovery kit board along with the default-provide
 
 ## Prerequisites
 
-- The STM32L496G-DISCO/BG96 board with a USB cable.
+- The STM32L496G-DISCO/BG96 board with a USB micro cable.
```
Comment (JZimnol) [inline docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md:7]: Same as above.

File: docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md [hunk 2]
```diff
@@ -4,15 +4,15 @@ Integrate your P-L496G-CELL02 Discovery kit board along with the default-provide
...
 - Installed **STM32CubeIDE**.
-- Installed **minicom** (for Linux) or RealTerm or PuTTy (for Windows) or other serial communication program.
+- The serial communication program, such as **minicom** (for Linux) or RealTerm or PuTTY (for Windows) installed.
 - A user with access to the Coiote IoT Device Management platform.
 
 ## Prepare binaries
 ### Use an already built binary
 
-To get the latest binary file and flash the board:
+To get the latest binary file and flash it onto the board:
```
Comment (JZimnol) [inline docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md:15]: Same as above.

File: docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md [hunk 3]
```diff
@@ -25,15 +25,17 @@ The board is now flashed: you can go to the [Connecting to the LwM2M Server](#co
...
+#### Cloning the Anjay freeRTOS client repository
 
-Enter the command line interface on your machine and paste the following command:
+Enter the command line interface on your machine and run the following command:
 
    ```
    git clone --recursive https://github.com/AVSystem/Anjay-freertos-client
    ```
+!!! Note
+    Only the simplest version of the Anjay Client is described here. To build binary with FOTA go to [Anjay Client with FOTA](#anjay-client-with-fota)
```
Comment (JZimnol) [inline docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md:36]: I don't know if it is required here. Also, you didn't add this to the L462 version.

File: docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md [hunk 4]
```diff
@@ -25,15 +25,17 @@ The board is now flashed: you can go to the [Connecting to the LwM2M Server](#co
...
 
-#### Part 2: Compiling the board
+#### Compiling the board
```
Comment (JZimnol) [inline docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md:38]: Looking at this... Do we really compile the board? We compile the application :D This can be changed in a different commit in this MR in all projects.

File: docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md [hunk 5]
```diff
@@ -46,8 +48,8 @@ Enter the command line interface on your machine and paste the following command
 0. In the Project Explorer, navigate to the **Anjay-freertos-client-STM32L496G-BG96** project:
     - Right-click on the project name and select **Build Project**. The build should take less than one minute to complete.
     - After the build is finished, right-click on the project name, select **Run As** and click the **1 STM32 Cortex-M C/C++ Application** option.
-        - In the **Lauch Configuration Selection**, choose the **Anjay-freertos-client-STM32L496G-BG96** option and click **OK**.
-0. After the build and run are complete, the board is now compiled.
+    - In the **Lauch Configuration Selection**, choose the **Anjay-freertos-client-STM32L496G-BG96** option and click **OK**.
```
Comment (JZimnol) [inline docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md:51]: I can't add a comment in the line `47`, but the "Import project" image should be updated.

File: docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md [hunk 6]
```diff
@@ -97,9 +99,86 @@ To connect the board:
...
 
-0. Go to Coiote DM to check if your device connected. Click **Next**, then **Go to Summary**, then **Finish**. You will see your Device Center view:
+0. Go to Coiote DM to check if your device is connected. Click **Next**, then **Go to Summary**, then **Finish**. You will see your Device Center view:
 
 ![Registered device](images/registered_device.png "Registered device")
 
 !!! tip
     LwM2M Server URI, endpoint name and other information can be found in the **Configuration** tab.
+
+## Anjay Client with FOTA
```
Comment (JZimnol) [inline docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md:109]:
```suggestion
## Anjay-freertos-client with FOTA (Firmware update Over the Air)
```

File: docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md [hunk 7]
```diff
@@ -97,9 +99,86 @@ To connect the board:
...
+
+Application can be built in basic version according to [Compiling the board](#compiling-the-board) section. Here compiling and flashing of extended version of Anjay Client is described. This version consists of **Secure Boot** and **Secure Firmware Update**.
```
Comment (JZimnol) [inline docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md:111]: Same comments as in the L462 chapter.

File: docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md [hunk 8]
```diff
@@ -97,9 +99,86 @@ To connect the board:
...
+in a secure way to prevent unauthorized updates and access to confidential on-device data such as code and
+firmware encryption key.
+
+The **Secure Boot** (Root of Trust services) is immutable code, always executed after a system reset, that checks STM32
+static protections, activates STM32 runtime protections and then verifies the authenticity and integrity of user
+application code before every execution in order to ensure that invalid or malicious code won't be run.
+
+### Additional prerequisites
+- **STM32CubeProgrammer** installed.
+- Python with the following modules: `pycryptodomex`, `ecdsa`, `numpy`, `pyelftools`.
```
Comment (JZimnol) [inline docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md:124]: Same comments as in the L462 chapter (about bash shell).

File: docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md [hunk 9]
```diff
@@ -97,9 +99,86 @@ To connect the board:
...
+   the protected environment. The binary is linked with the SBSFU code in step 2.
+
+0. Compile **SBSFU** application<br/>
+   This step compiles the SBSFU source code implementing the state machine and configuring the protections. In addition,
+   it links the code with the SECore binary generated at step 1 in order to generate a single SBSFU binary including the
+   SE trusted code.
+0. Compile **UserApp** application (set **Build configuration** to **Release**)<br/>
+   It generates:<br/>
+    1. The user application binary file that is uploaded to the device using the Secure Firmware Update process <br/>
+     (`Projects/STM32L496G-DISCO/UserApp/Binary/Anjay-freertos-client-STM32L496G-[MODEM].sfb`).
```
Comment (JZimnol) [inline docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md:141]: You can replace the `[MODEM]` with the `BG96`.

File: docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md [hunk 10]
```diff
@@ -97,9 +99,86 @@ To connect the board:
...
+
+!!! important
+    When flashed board with Secure Boot you will need to switch off secure protection to be able to flash the board again.
+
+To deactivate secure application please run **STM32_Programmer_CLI** (Program provided with STM32CubeProgrammer) tool with specific options:
+```
+./<path_to_STM32_Programmer_CLI> -c port=SWD mode=UR -ob RDP=0xBB -ob RDP=0xAA WRP1A_STRT=0xFF WRP1A_END=0x0 -ob displ
+```
+
+ ## Performing firmware update
```
Comment (JZimnol) [inline docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md:166]: Something went wrong here after rendering. I think that the space at the beginning is not needed.

Comment (JZimnol) [reply docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md:166]: Also, I would make it `###` in order to keep it under the `Anjay Client with FOTA` chapter.

File: docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md [hunk 11]
```diff
@@ -85,9 +85,76 @@ To connect the board:
...
+        You can set a custom firmware version in the `Application/Inc/default_config.h` file (using `FIRMWARE_VERSION` define).
+        It will be useful when performing FOTA to distinguish the firmware images from each other.
+
+### Flashing
+
+Use **STM32CubeProgrammer** application with `SBSFU_Anjay-freertos-client-B-L462E-CELL1-[MODEM].bin` file to program the board (it is advisable to perform **Full chip erase** first). You can open serial port to change default credentials in order to connect to Coiote DM.
+
+After that, you can use Coiote DM to perform firmware update with `Anjay-freertos-client-B-L462E-CELL1-[MODEM].sfb` file.
+
+## Performing firmware update
```
Comment (JZimnol) [inline docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md:142]: I would make it `###` in order to keep it under the `Anjay Client with FOTA` chapter.

File: docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md [hunk 12]
```diff
@@ -97,9 +99,86 @@ To connect the board:
...
+
+### Disable Secure Protection
+
+!!! important
+    When flashed board with Secure Boot you will need to switch off secure protection to be able to flash the board again.
+
+To deactivate secure application please run **STM32_Programmer_CLI** (Program provided with STM32CubeProgrammer) tool with specific options:
+```
+./<path_to_STM32_Programmer_CLI> -c port=SWD mode=UR -ob RDP=0xBB -ob RDP=0xAA WRP1A_STRT=0xFF WRP1A_END=0x0 -ob displ
+```
```
Comment (JZimnol) [inline docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md:164]: I would add it under one big "Important" section. Example (I haven't tried to render it):
```suggestion
!!! important
    When flashed board with Secure Boot you will need to switch off secure protection to be able to flash the board again. To deactivate secure application please run **STM32_Programmer_CLI** (Program provided with STM32CubeProgrammer) tool with specific options:
   ./<path_to_STM32_Programmer_CLI> -c port=SWD mode=UR -ob RDP=0xBB -ob RDP=0xAA WRP1A_STRT=0xFF WRP1A_END=0x0 -ob displ
```

I mean, don't create a separate `Disable Secure Protection` chapter.

**Review by JZimnol — COMMENTED** [review]
@jgoreckiavs first round of review done.

**Review by malw4 — COMMENTED** [review]
I added some comments in B-L462E-CELL1-TYPE1SC.md but it refers to the rest too. 
The titles are ok right now.

File: docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md [hunk 13]
```diff
@@ -85,9 +85,76 @@ To connect the board:
...
+The **X-CUBE-SBSFU Secure Boot and Secure Firmware Update** solution allows the update of the STM32 microcontroller built-in
+program with new firmware versions, adding new features and correcting issues. The update process is performed
+in a secure way to prevent unauthorized updates and access to confidential on-device data such as code and
+firmware encryption key.
+
+The **Secure Boot** (Root of Trust services) is immutable code, always executed after a system reset, that checks STM32
+static protections, activates STM32 runtime protections and then verifies the authenticity and integrity of user
+application code before every execution in order to ensure that invalid or malicious code won't be run.
+
+### Additional prerequisites
```
[the remark this answers is not in the document]
Comment (jgoreckiavs) [reply docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md:108]: I think this way it is clearer

File: docs/LwM2M_Client/Getting_started.md [hunk 14]
```diff
@@ -33,14 +33,6 @@ Integrate your P-L496G-CELL02 Discovery kit board along with the default-provide
 
 [Start integration](./STMicroelectronics/STM32L496G-DISCOBG96.md){: .md-button .md-button--big }
 
-### STM32L496G-DISCO/MONARCH
-
-Integrate your P-L496G-CELL02 Discovery kit along with the Sequans' Monarch GM01Q-STMOD expansion board using our Anjay LwM2M Client based on freeRTOS.
-
-![STM32L496G-DISCO/MONARCH](images/sqns.png "STM32L496G-DISCO/MONARCH")
-
```
Comment (Kucmasz) [inline docs/LwM2M_Client/Getting_started.md:41]: erherh

File: docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md [hunk 15]
```diff
@@ -97,9 +97,86 @@ To connect the board:
...
+0. Go to Coiote DM to check if your device is connected. Click **Next**, then **Go to Summary**, then **Finish**. You will see your Device Center view:
 
 ![Registered device](images/registered_device.png "Registered device")
 
 !!! tip
     LwM2M Server URI, endpoint name and other information can be found in the **Configuration** tab.
+
+## Anjay-freertos-client with FOTA (Firmware update Over the Air)
+
+Anjay application can be built in basic version (without FOTA) as described in the [Compiling the board](#compiling-the-board) section. In order to use FOTA, a few additional steps needs to be done, e.g. **Secure Boot** and **Secure Firmware Update** compilation.
```
Comment (malw4) [inline docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md:109]:
```suggestion
Anjay application can be built in basic version (without FOTA) as described in the [Compiling the board](#compiling-the-board) section. In order to use FOTA, a few additional steps need to be done, e.g. **Secure Boot** and **Secure Firmware Update** compilation.
```

File: docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md [hunk 16]
```diff
@@ -97,9 +97,86 @@ To connect the board:
...
+### Flash the board
+
+Use **STM32CubeProgrammer** application with `SBSFU_Anjay-freertos-client-STM32L496G-BG96.bin` file to program the board (it is advisable to perform **Full chip erase** first). You can open serial port to change default credentials in order to connect to Coiote DM.
+
+After that, you can use Coiote DM to perform firmware update with `Anjay-freertos-client-STM32L496G-BG96.sfb` file.
+
+!!! important
+    Disable Secure Protection
+
+    When flashed board with Secure Boot you will need to switch off secure protection to be able to flash the board again. To deactivate secure application please run **STM32_Programmer_CLI** (Program provided with STM32CubeProgrammer) tool with specific options:
```
Comment (malw4) [inline docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md:160]: Is this also an Important part? Because it looks quite strange 
image

File: docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md [hunk 17]
```diff
@@ -85,9 +85,79 @@ To connect the board:
...
+4. Go to Coiote DM to check if your device is connected. Click **Next**, then **Go to Summary**, then **Finish**. You will see your Device Center view:
 
 ![Registered device](images/registered_device.png "Registered device")
 
 !!! tip
     LwM2M Server URI, endpoint name and other information can be found in the **Configuration** tab.
+
+## Anjay-freertos-client with FOTA (Firmware update Over the Air)
+
+Anjay application can be built in basic version (without FOTA) as described in the [Compiling the board](#compiling-the-board) section. In order to use FOTA, a few additional steps needs to be done, e.g. **Secure Boot** and **Secure Firmware Update** compilation.
```
Comment (malw4) [inline docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md:97]:
```suggestion
Anjay application can be built in basic version (without FOTA) as described in the [Compiling the board](#compiling-the-board) section. In order to use FOTA, a few additional steps need to be done, e.g. **Secure Boot** and **Secure Firmware Update** compilation.
```

File: docs/LwM2M_Client/STMicroelectronics/B-U585I-IOT02A-BG96.md [hunk 18]
```diff
@@ -43,14 +43,14 @@ Enter the command line interface on your machine and paste the following command
...
-    - Right-click on the project name and select **Build Project**. The build should take less than one minute to complete.
+    - Right-click on the project name and select **Build Project**. Choose "Debug" configuration. The build should take less than one minute to complete.
     - After the build is finished, right-click on the project name, select **Run As** and click the **1 STM32 Cortex-M C/C++ Application** option.
-        - In the **Launch Configuration Selection**, choose the **Anjay-freertos-client-B-U585I-IOT02A-BG96** option and click **OK**.
-0. After the build and run steps are done, the board is now flashed and ready for next steps.
+    - In the **Launch Configuration Selection**, choose the **Anjay-freertos-client-B-U585I-IOT02A-BG96** option and click **OK**.
+0. After the build and run are complete, the board is flashed with compiled binary.
 
-## Connecting to the LwM2M Server
+## Connect to the LwM2M Server
```
Comment (malw4) [inline docs/LwM2M_Client/STMicroelectronics/B-U585I-IOT02A-BG96.md:53]: Line 60 change folowing to following

File: docs/LwM2M_Client/STMicroelectronics/B-U585I-IOT02A-BG96.md [hunk 19]
```diff
@@ -82,9 +82,9 @@ To connect the board:
     4. Click the **Add device** button and **Confirm** in the confirmation pop-up.
     5. In the **Connect your device** step, follow the next [section](#configuring-the-client) to run the client and connect it to the server.
 
-## Configuring the Client
+## Configure the Client
 
-1. With the board still connected to a serial port interface, open a serial communication program.
+1. With the board still connected to a serial port interface, open your serial communication program.
```
Comment (malw4) [inline docs/LwM2M_Client/STMicroelectronics/B-U585I-IOT02A-BG96.md:87]: Line 100 change truephone to Truephone

**Review by malw4 — COMMENTED** [review]
Please open `mkdocs serve` to check if everything looks as it should

**Review by JZimnol — COMMENTED** [review]
@jgoreckiavs second round of review done, my comments apply probably to both L462 and U5.

File: docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md [hunk 20]
```diff
@@ -4,16 +4,16 @@ Integrate your B-L462E-CELL1 Discovery kit board along with the TYPE 1SE module
...
-- Installed **minicom** (for Linux) or RealTerm or PuTTy (for Windows) or other serial communication program.
+- The serial communication program, such as **minicom** (for Linux) or RealTerm or PuTTY (for Windows) installed.
 - A user with access to the Coiote IoT Device Management platform.
 
 
 ## Prepare binaries
 ### Use an already built binary
 
-To get the latest binary file and flash the board:
+To get the latest binary file and flash it onto the board:
```
[the remark this answers is not in the document]
Comment (jgoreckiavs) [reply docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md:16]: I think that both versions are correct

File: docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md [hunk 21]
```diff
@@ -85,9 +85,79 @@ To connect the board:
...
+firmware encryption key.
+
+The **Secure Boot** (Root of Trust services) is immutable code, always executed after a system reset, that checks STM32
+static protections, activates STM32 runtime protections and then verifies the authenticity and integrity of user
+application code before every execution in order to ensure that invalid or malicious code won't be run.
+
+### Additional prerequisites
+- **STM32CubeProgrammer** installed.
+- Support for shell scripts execution (on Windows for example **Git** or **Cygwin** can be used).
+- [Python](https://www.python.org/downloads/) with the following modules: `pycryptodomex`, `ecdsa`, `numpy`, `pyelftools`.
```
[the remark this answers is not in the document]
Comment (malw4) [reply docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md:111]: But in places like nrfjprog or other additional programs, we add links. For me is safer and better to add this

File: docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md [hunk 22]
```diff
@@ -85,9 +85,79 @@ To connect the board:
...
+- Import B-L462E-CELL1_2_Images_SBSFU and B-L462E-CELL1_2_Images_SECoreBin projects from previously cloned repository to workspace.
```
[the remark this answers is not in the document]
Comment (jgoreckiavs) [reply docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md:112]: On the image all of them are selected to have the same image for each instruction. it was the same in the previous picture.
Instruction say to select only one project `In the **Projects** field, select **Anjay-freertos-client-B-L462E-CELL1-TYPE1SC** and click **Finish**`.

File: docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md [hunk 23]
```diff
@@ -85,9 +85,79 @@ To connect the board:
...
+     FW header <br/>
+     (`Projects/B-L462E-CELL1/UserApp/Binary/SBSFU_Anjay-freertos-client-B-L462E-CELL1-TYPE1SC.bin`).
+
+    !!! tip
+        You can set a custom firmware version in the `Application/Inc/default_config.h` file (using `FIRMWARE_VERSION` define).
+        It will be useful when performing FOTA to distinguish the firmware images from each other.
+
+### Flash the board
+
+Use **STM32CubeProgrammer** application with `SBSFU_Anjay-freertos-client-B-L462E-CELL1-TYPE1SC.bin` file to program the board (it is advisable to perform **Full chip erase** first). You can open serial port to change default credentials in order to connect to Coiote DM.
```
[the remark this answers is not in the document]
Comment (jgoreckiavs) [reply docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md:141]: I think that beginning user should be given with one way to complete each step to not mess around

File: docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md (hunk 21 above)
[the remark this answers is not in the document]
Comment (jgoreckiavs) [reply docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md:111]: Yeah, this is true that other program do not have links

File: docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md (hunk 23 above)
[the remark this answers is not in the document]
Comment (jgoreckiavs) [reply docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md:141]: Because `Prepare binaries` heading had no sense in previous version so I created separate section `Flash the board` and inserted drag-and-drop method there.

**Review by malw4 — DISMISSED** [review]
Looks good for me

File: docs/LwM2M_Client/STMicroelectronics/Building_LwM2M_applications/Building_LwM2M_applications_with_I-CUBE-Anjay.md [hunk 24]
```diff
@@ -113,7 +113,7 @@ Through Select Components menu choose desired components from the pack, in this
 - Board Support LwM2M
     - LwM2M Objects: P-L496G-CELL02
     - P-L496G-CELL02 BSP
-- Board Support X_STMOD_PLUS_MODEMS - BG96 or MONARCH (depending on your setup)
```
Comment (Kucmasz) [inline docs/LwM2M_Client/STMicroelectronics/Building_LwM2M_applications/Building_LwM2M_applications_with_I-CUBE-Anjay.md:116]: Actually I-CUBE-Anjay still supports MONARCH modem and will support it for a while, so there is no use in deleting it here.

**Review by Kucmasz — COMMENTED** [review]
@jgoreckiavs Reviewed.

File: docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md [hunk 25]
```diff
@@ -43,14 +39,17 @@ Enter the command line interface on your machine and paste the following command
...
-        - In the **Lauch Configuration Selection**, choose the **Anjay-freertos-client-B-L462E-CELL1-TYPE1SC** option and click **OK**.
-0. After the build and run are complete, the board is now compiled.
+    - Choose "Debug" configuration and build the project by right-clicking on the project name and selecting **Build Project**. The build should take less than one minute to complete.
+    - After the build is finished in `/Anjay-freertos-client-B-L462E-CELL1-TYPE1SC/Debug/` directory `Anjay-freertos-client-B-L462E-CELL1-TYPE1SC.bin` file should appear.
+
+## Flash the board
 
-## Connecting to the LwM2M Server
+0. To flash the board, open your **File manager** and drag the downloaded or compiled `.bin` file to your **DIS_L462RE** external device.
+0. You will see a blinking diode on your board. The diode will stop blinking as soon as the flashing is finished.
```
[the remark this answers is not in the document]
Comment (jgoreckiavs) [reply docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md:50]: My intentions was to move flashing board to another section to keep STM tutorials sections same as in other boards devzone tutorials. I think it is clearer to have shown only one way of flashing in flashing section.

[the remark this answers is not in the document]
Comment (Kucmasz) [reply docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md:50]: Disagree :( @malw4 pls prevent a fight and decide

[the remark this answers is not in the document]
Comment (malw4) [reply docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md:50]: Well, if we have another section for flashing so maybe it's a good idea to show two possibilities. But I agree with Janek that we want to have a correlation with other Devzone tutorials so maybe a small note about doing this in CubeIDE

File: docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md [hunk 26]
```diff
@@ -92,14 +90,93 @@ To connect the board:
...
+0. Go to Coiote DM to check if your device is connected. Click **Next**, then **Go to Summary**, then **Finish**. You will see your Device Center view:
 
 ![Registered device](images/registered_device.png "Registered device")
 
 !!! tip
     LwM2M Server URI, endpoint name and other information can be found in the **Configuration** tab.
+
+## Anjay-freertos-client with FOTA (Firmware update Over the Air)
+
+Anjay application can be built in basic version (without FOTA) as described in the [Build binary](#build-binary) and [Flash the board](#flash-the-board) section. In order to use FOTA, a few additional steps need to be done, e.g. **Secure Boot** and **Secure Firmware Update** compilation.
```
Comment (malw4) [inline docs/LwM2M_Client/STMicroelectronics/STM32L496G-DISCOBG96.md:107]: > in the [Build binary](#build-binary) and [Flash the board](#flash-the-board) section.

I think we don't have this sections

File: docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md [hunk 27]
```diff
@@ -80,14 +79,84 @@ To connect the board:
...
+4. Go to Coiote DM to check if your device is connected. Click **Next**, then **Go to Summary**, then **Finish**. You will see your Device Center view:
 
 ![Registered device](images/registered_device.png "Registered device")
 
 !!! tip
     LwM2M Server URI, endpoint name and other information can be found in the **Configuration** tab.
+
+## Anjay-freertos-client with FOTA (Firmware update Over the Air)
+
+Anjay application can be built in basic version (without FOTA) as described in the [Build binary](#build-binary) and [Flash the board](#flash-the-board) section. In order to use FOTA, a few additional steps need to be done, e.g. **Secure Boot** and **Secure Firmware Update** compilation.
```
Comment (malw4) [inline docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md:96]: > in the [Build binary](#build-binary) and [Flash the board](#flash-the-board) section.

Same here, we don't have this sections.

File: docs/LwM2M_Client/STMicroelectronics/B-L475E-IOT01A.md [hunk 28]
```diff
@@ -9,35 +9,35 @@ Integrate your B-L475E-IOT01A Discovery kit board.
...
-To get the latest binary file and flash the board:
+To get the latest binary file and flash it to the board:
 
 0. Go to [Anjay-zephyr-client](https://github.com/AVSystem/Anjay-zephyr-client/releases).
 0. Download the `demo_B-L475E-IOT01A1_merged.bin` file.
 0. To flash the board, open your **File manager** and drag the downloaded `.bin` file to your **DIS_L4IOT** external device.
 0. You will see a blinking diode on your board. The diode will stop blinking as soon as the flashing is finished.
 
-The board is now flashed: you can jump to the [Connecting to the LwM2M Server](#connecting-to-the-lwm2m-server) section.
+The board is now flashed: you can jump to the [Connect to the LwM2M Server](#connect-to-the-lwm2m-server) section.
```
Comment (malw4) [inline docs/LwM2M_Client/STMicroelectronics/B-L475E-IOT01A.md:22]: I don't like the ":" sign in this sentence. Also, if the steps are the same as in STM32L496G-DISCO/BG96 and B-L462E-CELL1/TYPE1SC why only in this tutorial do you use this sentence?

File: docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md [hunk 29]
```diff
@@ -4,37 +4,35 @@ Integrate your B-L462E-CELL1 Discovery kit board along with the TYPE 1SE module
...
+## Prepare Anjay client application
 ### Use an already built binary
 
-To get the latest binary file and flash the board:
+To get the latest binary file and flash it to the board:
 
 0. Go to [Anjay-freertos-client](https://github.com/AVSystem/Anjay-freertos-client/releases/).
 0. Download the `Anjay-freertos-client-B-L462E-CELL1-TYPE1SC.bin` file.
-0. To flash the board, open your **File manager** and drag the downloaded `.bin` file to your **DIS_L462RE** external device.
+0. To flash the board, open your **File manager** and drag the downloaded or compiled `.bin` file to your **DIS_L462RE** external device.
```
Comment (malw4) [inline docs/LwM2M_Client/STMicroelectronics/B-L462E-CELL1-TYPE1SC.md:20]: `or compiled`? why did you add this?

File: docs/LwM2M_Client/STMicroelectronics/B-L475E-IOT01A.md [hunk 30]
```diff
@@ -9,35 +9,35 @@ Integrate your B-L475E-IOT01A Discovery kit board.
...
 
 ### Start development using samples
 !!! Note
-    This step is optional. If you've gone through the [Use an already built binary](#use-an-already-built-binary) step, you can jump to [Connecting to the LwM2M Server](#connecting-to-the-lwm2m-server).
+    This step is optional. If you've gone through the [Use an already built binary](#use-an-already-built-binary) step, you can jump to [Connect to the LwM2M Server](#connect-to-the-lwm2m-server).
```
Comment (malw4) [inline docs/LwM2M_Client/STMicroelectronics/B-L475E-IOT01A.md:26]: If we add the 22nd line we can delete this note.

**Comment by JZimnol** [comment]
@jgorecki looks better, I'm accepting this MR.

**Review by JZimnol — APPROVED** [review]
